tawk_to.module

```diff
@@ -16,298 +15,247 @@ define('TAWK_TO_WIDGET_USER_ID', 'widget_user_id');
+    'page callback'    => 'tawk_to_admin_page',
+    'access arguments' => array('administer site configuration'),
+    'type'             => MENU_NORMAL_ITEM,
+  );
 
   $items['admin/config/tawk/setwidget'] = array(
-        'title'            => 'Set widget',
-        'page callback'    => 'tawk_to_set_widget',
-        'access arguments' => array('administer site configuration'),
-        'type'             => MENU_CALLBACK,
-      );
+    'title'            => 'Set widget',
+    'page callback'    => 'tawk_to_set_widget',
+    'access arguments' => array('administer site configuration'),
+    'type'             => MENU_CALLBACK,
+  );
 
   $items['admin/config/tawk/removewidget'] = array(
-        'title'            => 'Remove widget',
-        'page callback'    => 'tawk_to_remove_widget',
-        'access arguments' => array('administer site configuration'),
-        'type'             => MENU_CALLBACK,
-      );
-
-  $items['admin/config/tawk/setoptions'] = array(
-        'title'            => 'Set widget options',
-        'page callback'    => 'tawkto_widget_options',
-        'access arguments' => array('administer site configuration'),
-        'type'             => MENU_CALLBACK,
-      );
+    'title'            => 'Remove widget',
+    'page callback'    => 'tawk_to_remove_widget',
+    'access arguments' => array('administer site configuration'),
+    'type'             => MENU_CALLBACK,
+  );
 
   return $items;
 }
 
 /**
- * Displays widget configuration iframe.
- *
- * Configuration iframe contains functionality from
- * tawk.to server which is responsible for verifying
- * user login details and also retrieves the user's 
- * property and widget list and comunicates back to the
- * parent page so that changes can be saved in the config files.
+ * Implements tawk.to module admin page
  */
-function tawkto_widget_customization()
-{
-    $base_url = tawk_to_get_base_url();
-    $iframe_url = tawk_to_get_iframe_url();
-    $widget = tawk_to_get_widget();
-
-    global $user;
-    $sameUser = true;
-    $value = config_get(TAWK_TO_CONFIG_NAME, TAWK_TO_WIDGET_USER_ID);
-    if (!empty($value) && $user->uid!=$value) {
-        $sameUser = false;
-    }
-
-    $displayOpts = config_get(TAWK_TO_CONFIG_NAME, TAWK_TO_WIDGET_OPTIONS);
-    if ($displayOpts && !empty($displayOpts)) {
-        $displayOpts = json_decode($displayOpts);
-    } else {
-        $displayOpts = null;
-    }
-
-    $module_path = backdrop_get_path('module', 'tawk_to');
-    backdrop_add_css($module_path . '/css/tawk_to.admin.css', 'file');
-
-    return tawk_to_render_widget_iframe($base_url, $iframe_url, $widget, $displayOpts, $sameUser);
+function tawk_to_admin_page() {
+  global $user;
+  $sameUser = true;
+  $value = config_get(TAWK_TO_CONFIG_NAME, TAWK_TO_WIDGET_USER_ID);
+
+  if (!empty($value) && $user->uid!=$value) {
+    $sameUser = false;
+  }
+
+  $visibility_opts_form = backdrop_get_form('tawk_to_visibility_opts_form');
+
```

Comment:
This is where `backdrop_add_css()` should be called to add the stylesheet for the admin page (rather than in the template file).